File: Top 100 Codes/03-Codes for Recursion/03-Largest element in an array/Largestelementinanarray.py
def largest_element(arr, n):
    """Recursive function to find the largest element in an array"""
    if n == 1:  # Base condition: If only one element, return it
        return arr[0]
    
    return max(arr[n - 1], largest_element(arr, n - 1))  # Compare last element with recursive result

# Taking user input for array size
n = int(input("Enter the number of elements: "))

# Taking user input for array elements
arr = list(map(int, input("Enter the elements: ").split()))

# Checking if the array is empty
if n == 0:
    print("Array is empty, no largest element.")
else:
    # Display the largest element
    print("Largest Element is:", largest_element(arr, n))

File: Top 100 Codes/03-Codes for Recursion/03-Largest element in an array/Largestelementinanarray.cpp
#include <bits/stdc++.h>
using namespace std;

// Recursive function to find the largest element in an array
int largest_element(int n, int arr[]) {
    if (n == 1)  // Base condition: If only one element, return it
        return arr[0];

    return max(arr[n - 1], largest_element(n - 1, arr)); // Compare last element with recursive result
}

int main() {
    int n;
    
    // Taking user input for array size
    cout << "Enter the number of elements: ";
    cin >> n;

    int arr[n]; // Declare array
    
    // Taking user input for array elements
    cout << "Enter " << n << " elements: ";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    // Display the largest element
    cout << "Largest Element is: " << largest_element(n, arr) << endl;

    return 0;
}
